vehicle/vw/id/provider.go

```diff
@@ -32,15 +33,12 @@ var _ api.Battery = (*Provider)(nil)
 // SoC implements the api.Vehicle interface
 func (v *Provider) SoC() (float64, error) {
 	res, err := v.statusG()
-	if err == nil {
-		err = res.Error.Extract("batteryStatus")
-	}
 
-	if err == nil {
-		return float64(res.Data.BatteryStatus.CurrentSOCPercent), nil
+	if err == nil && res.Charging != nil {
+		return float64(res.Charging.BatteryStatus.Value.CurrentSOCPct), nil
 	}
 
-	return 0, err
+	return 0, fmt.Errorf("SoC not avaliable")
```

Comment:
Bitte Ursprungserror zurück geben. Der geht sonst unter.

Reply:
`return 0, fmt.Errorf("SoC not avaliable: %s", err)` <- ich würde das hier vorschlagen um zu dem generischen error Kontext hinzuzufügen.

Aktuell ist der angezeigte Fehler bei `evcc vehicle` zb. `.SoC:          unexpected status: 404` dann wäre es `SoC not avaliable: unexpected status: 404`

Reply:
Bitte einfach so lassen.